mongoengine/connections_manager.py

```diff
@@ -0,0 +1,110 @@
+from collections import defaultdict
+
+from mongoengine.connection import get_connection, get_db
+
+__all__ = ['InvalidCollectionError', 'connection_manager']
+
+
+class InvalidCollectionError(Exception):
+    pass
+
+
+class ConnectionManager(object):
+    connections_registry = defaultdict(dict)
+
+    def get_and_setup(self, doc_cls, alias=None, collection_name=None):
```

Comment:
This could use a docstring, especially as a public method. What it doesn, when it should be used, etc.